torchvision/datasets/sun397.py

```diff
@@ -0,0 +1,108 @@
+
+    def __init__(
+        self,
+        root: str,
+        split: str = "train",
+        partition: Optional[int] = 1,
+        download: bool = True,
+        transform: Optional[Callable] = None,
+        target_transform: Optional[Callable] = None,
+    ) -> None:
+        super().__init__(root, transform=transform, target_transform=target_transform)
+        self.split = verify_str_arg(split, "split", ("train", "test"))
+        self.partition = partition
+        self.data_dir = Path(self.root) / "SUN397"
+
+        if self.partition is not None:
+            if self.partition < 0 or self.partition > 10:
+                raise RuntimeError("Enter a valid integer partition from 1 to 10 or None, for entire dataset")
+
+        if download:
+            self._download()
+
+        if not self._check_exists():
+            raise RuntimeError("Dataset not found. You can use download=True to download it")
+
+        with open(self.data_dir / "ClassName.txt", "r") as f:
+            self.classes = [c[3:].strip() for c in f]
+
+        self.class_to_idx = dict(zip(self.classes, range(len(self.classes))))
+
+        self._labels = []
+        self._image_files = []
+        if self.partition is not None:
+            with open(self.data_dir / f"{self.split.title()}ing_{self.partition:02d}.txt", "r") as f:
+                self._image_files, self._labels = zip(
+                    *(
+                        (
+                            self.data_dir.joinpath(*posix_rel_path.split("/")),
+                            self.class_to_idx["/".join(Path(posix_rel_path).parts[1:-1])],
+                        )
+                        for posix_rel_path in (line.strip()[1:] for line in f)
+                    )
+                )
+
+        else:
+            for path, _, files in os.walk(self.data_dir):
+                for file in files:
+                    if file[:3] == "sun":
+                        self._image_files.append(Path(path) / file)
+                        self._labels.append(Path(path).relative_to(self.data_dir).as_posix()[2:])
+
+    def __len__(self) -> int:
+        return len(self._image_files)
+
+    def __getitem__(self, idx) -> Tuple[Any, Any]:
+        image_file, label = self._image_files[idx], self._labels[idx]
+        image = PIL.Image.open(image_file).convert("RGB")
+
+        if self.transform:
+            image = self.transform(image)
+
+        if self.target_transform:
+            label = self.target_transform(label)
+
+        return image, label
+
+    def _check_exists(self) -> bool:
+        return self.data_dir.exists() and self.data_dir.is_dir()
+
+    def extra_repr(self) -> str:
+        return "Split: {split}".format(**self.__dict__)
+
+    def _download(self) -> None:
+        if self._check_exists:
+            return
+        download_and_extract_archive(self._URL, download_root=self.root, md5=self._MD5)
+        download_and_extract_archive(self._PARTITIONS_URL, download_root=self.data_dir)
```

Comment:
`download_and_extract_archive` works with `Path`'s, but is only annotated for `str`'s.

```suggestion
        download_and_extract_archive(self._PARTITIONS_URL, download_root=str(self.data_dir))
```